src/Engine/MapController.java

```diff
@@ -153,39 +154,39 @@ private void handleFreeTileSelect(InputHandler.InputAction input)
         myGame.moveCursorRight();
         break;
       case SEEK: // Move the cursor to either the next unit that is ready to move, or an owned usable property.
-        boolean found = false;
-        int tries = 0;
-        int numUnits = myGame.activeCO.units.size();
-        ArrayList<XYCoord> usableProperties = Utils.findUsableProperties(myGame.activeCO, myGame.gameMap);
-        int numFreeIndustries = usableProperties.size();
-        int maxTries = numUnits + numFreeIndustries;
-        while ((maxTries > 0) && !found && (tries < maxTries))
+        
+        if( null == seekLocations )
         {
-          // Normalize the index to allow wrapping.
-          if( nextSeekIndex >= maxTries )
-          {
-            nextSeekIndex = 0;
-          }
+          // Populate our list of seek candidates and sort them by distance from the cursor.
+          nextSeekIndex = 0; // We are going to rebuild the list; start from the start.
+
+          // First get all active units, sorted.
+          ArrayList<XYCoord> unitLocations = new ArrayList<XYCoord>();
+          unitLocations.addAll(Utils.findLocationsNearUnits(myGame.gameMap, myGame.activeCO.units, 0));
+          unitLocations.removeIf(xy -> myGame.gameMap.getLocation(xy).getResident().isTurnOver);
+          Utils.sortLocationsByDistance(myGame.getCursorCoord(), unitLocations);
+
+          // Find all usable properties, sorted.
+          ArrayList<XYCoord> usableProperties = Utils.findUsableProperties(myGame.activeCO, myGame.gameMap);
+          usableProperties.removeIf(xy -> myGame.gameMap.getLocation(xy).getResident() != null);
+          Utils.sortLocationsByDistance(myGame.getCursorCoord(), usableProperties);
```

Comment:
Om nom delicious lambda

Reply:
`Utils.sortLocationsByDistance` is a perfectly standard function, I assure you

Reply:
I didn't wanna split the code block. I was referring to 171.

Reply:
Yes.